python_modules/dagster/dagster_tests/storage_tests/test_input_manager.py

```diff
@@ -379,6 +383,43 @@ def check_input_managers():
     check_input_managers.execute_in_process()
 
 
+def test_input_manager_with_assets():
+    @asset
+    def upstream() -> int:
+        return 1
+
+    @asset(ins={"upstream": AssetIn(input_manager_key="special_io_manager")})
+    def downstream(upstream) -> int:
+        return upstream + 1
+
+    class MyIOManager(IOManager):
+        def load_input(self, context):
+            assert context.upstream_output is not None
+
+            return 2
+
+        def handle_output(self, context, obj):
+            ...
+
+    group = AssetGroup(
```

Comment:
AssetGroups are a deprecated concept. Instead, you can call `materialize` on a list of assets.

Reply:
I tried that but it didn't work the way i needed it to. Here's the old code snippet 
```
def test_loading_already_materialized_asset():
    with instance_for_test() as instance:
        # materialize the upstream
        materialize([upstream], instance=instance)

        # materialize just the downstream
        materialize(
            [downstream],
            resources={
                "special_io_manager": IOManagerDefinition.hardcoded_io_manager(MyIOManager())
            },
            instance=instance,
        )
```

but that would give an error when materializing `downstream` that the dependency `upstream` didn't exist. Putting both assets in a group and then materializing via `selection` was the only way i could replicate the bug. if there's a better way to do this, i'm happy to change it

Reply:
You can do 

```python
materialize(
            [*upstream_asset.to_source_assets(), downstream],
```